### DISCUSSION TITLE
How can I show source TEXT_PLACEHOLDER?

### DISCUSSION THREAD

**Discussion by morphinestyle** [akeeba/panopticon#772] [Q&A (Community Support)] [answered]
How can I show source TEXT_PLACEHOLDER on Panopticon during translations, instead of viewing text strings in any language?

**Comment by nikosdion** [comment]
You cannot. It will not be implemented either.

**Comment by morphinestyle** [comment, marked as answer]
Well, since there is no way or intention to implement it, I solved it by converting the entire panopticon.pot to a "system language" containing the PLACEHOLDER.

![pano5](https://github.com/user-attachments/assets/b9423c48-229f-4fd9-affe-ed1b2f1f9c1e)

![pano6](https://github.com/user-attachments/assets/ad0f2c86-5c2c-40fe-be5c-30957bd80e4d)

![pano7](https://github.com/user-attachments/assets/8bf7a6bb-9b74-41f3-8413-ebba2889517a)

**Reply by nikosdion** [reply]
That's a perfectly valid way to do this. If you play around with the PO to INI conversion scripts in the repo you can even create a script to automatically extract all the keys and output them to an INI file in the format `KEY="KEY"` (double quotes can make the difference, depending on the PHP version, so make sure to include them).

BTW, the reason I cannot implement that as a core feature is a chicken and egg issue. Controlling whether to do that would require reading the configuration before loading the languages, which is not possible for some technical reasons that have to do with the way we detect languages and interact with configuration stored in environment variables. It's... complicated.

If I find a different solution I'll let you know -- but for now your way works best.

**Reply by morphinestyle** [reply]
Sorry Nicholas, but unfortunately I didn't do it with a PO-to-INI conversion script, but semi-manually with other software on the local computer.
